### DISCUSSION TITLE
Misleading example of `inplace` option in the doc

### DISCUSSION THREAD

**Discussion by stla** [pyvista/pyvista#2077] [Q&A]
Hello,

  On [this page](https://docs.pyvista.org/api/core/_autosummary/pyvista.ExplicitStructuredGrid.translate.html), the example contains this line of code:

`trans = mesh.translate((2, 1, 2), inplace=True)`

Maybe I misunderstand `inplace` but I don't think so. By setting `inplace=True`, the mesh `mesh` is transformed by the translation, no? Then `inplace=False` would be more appropriate.

**Comment by MatthewFlamm** [comment]
I agree that this example is misleading.  In the example, `mesh is trans` evaluates to `True`, but I don't think we want to encourage this pattern of use.  The only acceptable pattern of use I can think of is:

```py
mesh.translate((2, 1, 2), inplace=True).rotate_y(90, inplace=True)
mesh.center
```

**Reply by banesullivan** [reply]
I agree this is confusing and I am likely the culprit with my PR #1860 that refactored a ton of these examples... at the time my goal was to make sure everything worked, not necessarily to make everything perfect. I think It would be good to go back through all of these examples and make sure they make sense/are intuitive. Perhaps the best time for that would be when we handle the default `inplace=True` deprecation for the `PolyData` filters

However, this does demonstrate a recent feature of the API where now ALL calls to a filter always have a return type - see #1296. when updating in place, the in-place edited mesh's reference is returned (though not needed). Originally, I was not in favor of this and propose #1861 to revert it, but since then, I have seen its value and think there are scenarios where this is useful.
